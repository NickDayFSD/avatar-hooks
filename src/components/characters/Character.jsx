import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Character = ({ _id, name, photoUrl, affiliation }) => (
  <Link to={`/${_id}`}>
    <figure>
      <img src={photoUrl} alt={name} />
      <figcaption>
        <p>{name}</p>
        <p>{affiliation}</p>
      </figcaption>
    </figure>
  </Link>
);

Character.propTypes = {
  _id: PropTypes.string,
  name: PropTypes.string,
  photoUrl: PropTypes.string,
  affiliation: PropTypes.string,
};

export default Character;
