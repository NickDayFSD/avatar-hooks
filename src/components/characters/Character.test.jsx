/* eslint-disable max-len */
import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Character from './Character';

describe('Character component', () => {
  it('renders a Character', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Character
          name="Aang"
          photoUrl="https://vignette.wikia.nocookie.net/avatar/images/a/ae/Aang_at_Jasmine_Dragon.png/revision/latest?cb=20130612174003"
          affiliation="Air Acolytes Air Nomads Air Scouts (formerly) Team Avatar"
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
