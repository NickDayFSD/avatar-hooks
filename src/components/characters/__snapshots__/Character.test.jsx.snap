// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a Character 1`] = `
<DocumentFragment>
  <a
    href="/undefined"
  >
    <figure>
      <img
        alt="Aang"
        src="https://vignette.wikia.nocookie.net/avatar/images/a/ae/Aang_at_Jasmine_Dragon.png/revision/latest?cb=20130612174003"
      />
      <figcaption>
        <p>
          Aang
        </p>
        <p>
          Air Acolytes Air Nomads Air Scouts (formerly) Team Avatar
        </p>
      </figcaption>
    </figure>
  </a>
</DocumentFragment>
`;
